src/software/ai/hl/stp/tactic/goalie/goalie_fsm.cpp

```diff
@@ -111,6 +112,40 @@ Point GoalieFSM::findGoodChipTarget(
     return chip_target;
 }
 
+bool GoalieFSM::shouldEvacuateCrease(const Update &event)
+{
+    Rectangle friendly_defense_area =
+        event.common.world_ptr->field().friendlyDefenseArea();
+    Ball ball = event.common.world_ptr->ball();
+
+    // calculate inflated crease obstacle
+    double robot_radius_expansion_amount =
+        ROBOT_MAX_RADIUS_METERS *
+        robot_navigation_obstacle_config.robot_obstacle_inflation_factor();
+    Rectangle inflated_defense_area =
+        friendly_defense_area.expand(robot_radius_expansion_amount);
+
+    bool ball_in_dead_zone =
+        !contains(friendly_defense_area, ball.position()) &&
+        contains(friendly_defense_area.expand(robot_radius_expansion_amount),
+                 ball.position());
+
+    // goalie should only evacuate crease if there are no enemy robots nearby
+    double safe_distance_multiplier = goalie_tactic_config.safe_distance_multiplier();
+    double nearest_enemy_distance_to_ball = distance(
```

Comment:
you should account for the enemy velocity to figure out whether it's safe to intercept

Reply:
What kind of formula are u thinking? I thought about comparing the time it takes for the goalie/enemy to reach the ball but i don't think that would work if say the enemy robot was closer but had velocity 0

Reply:
oh I missed this comment. Why wouldn't time work?

Reply:
I was thinking of the edge case where the nearest enemy robot is stationary (velocity 0). Then it wouldn't work to get distance divided by velocity to get time. I guess an option would be to handle that specific edge case by comparing distance from ball instead of time to reach ball?

Reply:
we could use the new trajectory planner here right @nimazareian. also this PR is good to go, so feel free to merge